docs/reference/query-builders.md

```diff
@@ -0,0 +1,382 @@
+
+Since this is the main way additional criteria get merged, this method performs a few transformations of the query to make it more readable when `APOS_LOG_ALL_QUERIES=1` is in the environment.
+
+### `archived()`
+
+```
+query.archived(false)
+```
+
+The `archived` builder tells the query whether or not to include archived documents in the results. It takes boolean value, `undefined`, or `null` (empty).
+
+If the builder value is `false`, `undefined` or this method is never called, the query only returns docs that are *not* in the archive. This is the default behavior for document queries.
+
+If the value is `true` the query returns *only* docs in the archive. Note that permissions would still prevent a typical site visitor from obtaining any results, but an editor might.
+
+If the value is `null` (not `undefined`), the query returns docs regardless of archived status.
+
+### `areas()`
+
+```
+query.areas(false)
+
+query.areas([ 'thumbnail' ])
+```
+
+The `areas` builder controls whether to call the `load` methods of widget type managers for widgets in areas. `load` methods are used to do things such as populating relationships in widgets and other asynchronous actions when areas are loaded.
+
+The default value is `true`. Pass `false` to prevent the extra processing for a particular query. This is helpful to make queries faster when detailed area data is not needed.
+
+The value can also be an array containing strings representing area field names or dot paths to nested area fields. For example, with `.areas([ 'thumbnail' ])` you could load only the `'thumbnail'` area for all pages matching the query.
+
+### `attachments()`
+
+```
+query.attachments(true)
+```
+
+Passing `true` to the `attachments` builder will annotate all attachment fields in the returned documents with URLs and other metadata (primarily for images). This uses the the `apos.attachment.all` method with the `annotate: true` option. The builder defaults to `false`.
+
+### `autocomplete()`
+
+```
+query.autocomplete('tree house')
+```
+
+The `autocomplete` builder operates as a sort of fuzzy search for documents. It accepts a string and uses document search text properties to find matches. The string can contain multiple words, but each word (or segment) needs to be a partial string to find non-exact results. So "tree" will find results with "treehouse," but "treehouses" will not find "treehouse" (it is not meant as an advanced search tool, but can help find similar results).
+
+### `choices()`
+
+```
+query.choices('color, species')
+
+query.choices(['color', 'species'])
+```
+
+The `choices` builder is use to populate a `choices` object on the query results based on the filters passed to it. Filters can be passed in as a single string of comma-separated values or an array of strings.
```

Comment:
This is incorrect re: how to get the results.

After awaiting `query.toArray`, you can invoke `query.get('choicesResults')` to get an object in which each property is the name of a field you passed to `choices` and each value is an array of choice objects with `label` and `value` properties.

This may seem a bit awkward but for most server side applications you would use `toChoices` or just `toDistinct`. This query builder was designed to help the `getAll` REST API populate its `choices` property.

Reply:
It's not true that you can pass a string, that only works when "launder" is called, which only happens in the REST API (more specifically when `applyBuildersSafely` is used to apply the builders). We don't impose the performance hit of launder calls in pure server side programming, which is what we're creating a reference for here...

Well, sort of. Actually, since this is the reference for builders and some builders have a `launder` function, we *should* be talking about that and documenting what it accepts. But we have to make clear it's only applicable when the builder in question is present in the query string of a REST API request, or when `applyBuildersSafely` is used in a custom API.

Reply:
It sounds like this is not worth documenting since it's especially complicated and there are better ways for most code to do the same thing. What do you think?